lib/minitest.rb

```diff
@@ -592,6 +594,23 @@ def summary # :nodoc:
       "%d runs, %d assertions, %d failures, %d errors, %d skips%s" %
         [count, assertions, failures, errors, skips, extra]
     end
+
+    def failed_tests # :nodoc:
+      lines = [ "\n# Failed tests:\n" ]
+
+      results.reject(&:skipped?).each do |result|
+        file_path = relative_from_pwd(result.failure.location.sub(/:\d+$/, ""))
+        lines << "ruby #{file_path} -n #{result.class}##{result.name}"
+      end
+
+      lines.join("\n") + "\n"
+    end
```

Comment:
Note: This can be also written more functionally like:

```ruby
    def failed_tests
      results.reject(&:skipped?).map { |result|
        file_path = relative_from_pwd(result.failure.location.sub(/:\d+$/, ""))
        "ruby #{file_path} -n #{result.class}##{result.name}"
      }.unshift("\n# Failed tests:\n").join("\n") + "\n"
    end
```

…but it’s IMHO less readable.